Source/WebGPU/WGSL/AttributeValidator.cpp

```diff
@@ -305,7 +305,8 @@ void AttributeValidator::validateAlignment(const SourceSpan& span, AddressSpace
 
             validateAlignment(member.span(), addressSpace, type);
 
-            if (member.offset() % requiredAlignment(type))
+            auto alignment = requiredAlignment(type);
+            if ((member.offset() % alignment) || (member.alignment() % alignment))
                 error(member.span(), "offset of struct member "_s, structure.name(), "::"_s, member.name(), " must be a multiple of "_s, String::number(requiredAlignment(type)), " bytes, but its offset is "_s, String::number(member.offset()), " bytes"_s);
```

Comment:
Does this message need to be updated to say, "offset and alignment of struct member ...".

Even better would be to separate the two tests for `member.offset()` and `member.alignment()` to give the specific error message, "offset of ..." or "alignment of ..."

Reply:
Made two separate messages, thanks for the suggestion 👍